### PR TITLE
Performance/memory optimizations

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

File: http/src/main/java/io/micronaut/http/MediaType.java [hunk 1]
```diff
@@ -446,13 +468,64 @@ public MediaType(String name, String extension, Map<String, String> params) {
...
             }
         }
-        if (params != null) {
-            parameters.putAll(params);
-        }
-
         this.strRepr = toString0();
     }
 
+    public static MediaType of(String name) {
```
Comment (graemerocher) [inline http/src/main/java/io/micronaut/http/MediaType.java]: Needs javadoc. Also parameter and return type should be annotated with `@NonNull`

File: http/src/main/java/io/micronaut/http/MediaType.java [hunk 2]
```diff
@@ -587,8 +660,14 @@ private String toString0() {
         if (parameters.isEmpty()) {
             return name;
         } else {
-            return name + ";" + parameters.entrySet().stream().map(Object::toString)
-                    .collect(Collectors.joining(";"));
+            StringBuilder sb = new StringBuilder(name);
+            for (Map.Entry<CharSequence, String> e : parameters.entrySet()) {
```
Comment (graemerocher) [inline http/src/main/java/io/micronaut/http/MediaType.java]: Would `parameters.forEach((key, value))` not be more efficient? As far as I understand that removes the need to create the  `Entry` instances

File: http/src/main/java/io/micronaut/http/MutableHttpHeaders.java [hunk 3]
```diff
@@ -37,6 +37,8 @@
  */
 public interface MutableHttpHeaders extends MutableHeaders, HttpHeaders  {
 
+    ZoneId GMT = ZoneId.of("GMT");
```
Comment (graemerocher) [inline http/src/main/java/io/micronaut/http/MutableHttpHeaders.java:43]: needs javadoc

**Review by graemerocher — CHANGES_REQUESTED** [review]
Overall looks good, just a few small changes

File: http/src/main/java/io/micronaut/http/MediaType.java (hunk 2 above)
Comment (dstepanov) [reply http/src/main/java/io/micronaut/http/MediaType.java]: Yeah good catch

**Review by graemerocher — APPROVED** [review]

**Pull request merged**